src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -0,0 +1,220 @@
+
+    bool use_distinct_data_transfer_queue = false;
+
+    bool use_one_queue_for_graphics_and_present = false;
+
+    if (queue_family_index_for_both_graphics_and_presentation) {
+        spdlog::debug("One queue for both graphics and presentation will be used.");
+
+        graphics_queue_family_index = queue_family_index_for_both_graphics_and_presentation.value();
+        present_queue_family_index = graphics_queue_family_index;
+
+        use_one_queue_for_graphics_and_present = true;
+
+        // For now, we only need one queue family.
+        std::uint32_t number_of_combined_queues_to_use = 1;
+
+        // In this case, there is one queue family which can be used for both graphics and presentation.
+        VkDeviceQueueCreateInfo create_info = {};
+        create_info.sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
+        create_info.queueFamilyIndex = queue_family_index_for_both_graphics_and_presentation.value();
+        create_info.queueCount = number_of_combined_queues_to_use;
+        create_info.pQueuePriorities = &::default_queue_priority;
+
+        queues_to_create.push_back(create_info);
+    } else {
+        spdlog::debug("No queue found which supports both graphics and presentation.");
+        spdlog::debug("The application will try to use 2 separate queues.");
+
+        // We have to use 2 different queue families.
+        // One for graphics and another one for presentation.
+
+        // Check which queue family index can be used for graphics.
+        graphics_queue_family_index = settings_decision_maker.find_graphics_queue_family(graphics_card);
+
+        if (!graphics_queue_family_index) {
+            throw std::runtime_error("Could not find suitable queue family indices for graphics!");
+        }
+
+        // Check which queue family index can be used for presentation.
+        present_queue_family_index = settings_decision_maker.find_presentation_queue_family(graphics_card, surface);
+
+        if (!present_queue_family_index) {
+            throw std::runtime_error("Could not find suitable queue family indices for presentation!");
+        }
+
+        spdlog::debug("Graphics queue family index: {}.", graphics_queue_family_index.value());
+        spdlog::debug("Presentation queue family index: {}.", present_queue_family_index.value());
+
+        // For now, we only need one queue family.
+        std::uint32_t number_of_graphics_queues_to_use = 1;
+
+        // Set up one queue for graphics.
+        VkDeviceQueueCreateInfo create_info = {};
+        create_info.sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
+        create_info.queueFamilyIndex = graphics_queue_family_index.value();
+        create_info.queueCount = number_of_graphics_queues_to_use;
+        create_info.pQueuePriorities = &::default_queue_priority;
+
+        queues_to_create.push_back(create_info);
+
+        // For now, we only need one queue family.
+        std::uint32_t number_of_present_queues_to_use = 1;
```

Comment:
Inline or constexpr.

Reply:
I inline this and `number_of_queues_to_use`.

Reply:
Done.